app/models/skin.rb

```diff
@@ -188,6 +188,7 @@ def cache!
     self.public = true
     self.official = true
     save!
+    Rails.cache.increment('skins_generation/'+(self.type.nil? ? ("site_skin") : self.id.to_s) )
```

Comment:
I mean, Rails uses metaprogramming all over the place to make more intuitive and readable code, which is significantly more expensive than the simple cost of a method call. Also a quick google finds that method dispatch is not a bottleneck for Rails so I would not let this be a bar.